lib/addons/ebs-csi-driver/index.ts

```diff
@@ -1,49 +1,130 @@
+  saName: "ebs-csi-controller-sa",
+  storageClass: "gp2", // Set the default StorageClass to gp3
 };
 
 /**
  * Implementation of EBS CSI Driver EKS add-on
  */
 @supportsALL
 export class EbsCsiDriverAddOn extends CoreAddOn {
+  readonly ebsProps: EbsCsiDriverAddOnProps;
+
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+        addOnName: defaultProps.addOnName,
+        version: options?.version ?? defaultProps.version,
+        saName: defaultProps.saName
+
+    });
+
+    this.ebsProps = {
+        ...defaultProps,
+        ...options,
+    };
+}
 
-    constructor(readonly options?: EbsCsiDriverAddOnProps) {
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName
-        });
-    }
 
-    providePolicyDocument(clusterInfo: ClusterInfo) : PolicyDocument {
-        return getEbsDriverPolicyDocument(clusterInfo.cluster.stack.partition, this.options?.kmsKeys);
+  providePolicyDocument(clusterInfo: ClusterInfo): PolicyDocument {
+    return getEbsDriverPolicyDocument(
+      clusterInfo.cluster.stack.partition,
+      this.options?.kmsKeys
+    );
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+    let updateSc: KubernetesManifest;
+
+    if (this.ebsProps.storageClass) {
+      // patch resource on cluster
+      const patchSC = new KubernetesPatch(
+        cluster.stack,
+        `${cluster}-RemoveGP2SC`,
+        {
+          cluster: cluster,
+          resourceName: "storageclass/gp2",
+          applyPatch: {
+            metadata: {
+              annotations: {
+                "storageclass.kubernetes.io/is-default-class": "false",
+              },
+            },
+          },
+          restorePatch: {
+            metadata: {
+              annotations: {
+                "storageclass.kubernetes.io/is-default-class": "true",
+              },
+            },
+          },
+        }
+      );
+
+      // Create and set gp3 StorageClass as cluster-wide default
+      updateSc = new KubernetesManifest(cluster.stack, `${cluster}-SetDefaultSC`, {
+        cluster: cluster,
+        manifest: [
+          {
+            apiVersion: "storage.k8s.io/v1",
+            kind: "StorageClass",
+            metadata: {
+              name: "gp3",
+              annotations: {
+                "storageclass.kubernetes.io/is-default-class": "true",
+              },
+            },
+            provisioner: "ebs.csi.aws.com",
+            reclaimPolicy: "Delete",
+            volumeBindingMode: "WaitForFirstConsumer",
+            parameters: {
+              type: "gp3",
+              fsType: "ext4",
+              encrypted: "true",
+            },
+          },
+        ],
+      });
+
+      const updateScConstruct = new scConstruct(cluster.stack, `${cluster}-PatchSC`, updateSc);
+      return Promise.resolve(updateSc);

```

Comment:
you override the construct that we return. That means nobody will be able to depend on the ebs addon, since you return the patch. The addon must return the CfnAddon construct.

Reply:
Updated code to perform core add-on deployment first and added that as dependency for patch and new manifest deployment. Updated the return as well. Please let me know if this aligns with the design. 

lint and test succeeded locally:
```
Test Suites: 14 passed, 14 total
Tests:       68 passed, 68 total
Snapshots:   0 total
Time:        28.743 s
```